libs/posthog_client_lite/src/lib.rs

```diff
@@ -0,0 +1,616 @@
+                        "Can not parse the right side of the condition as a number",
+                    ));
+                };
+                // Left can either be a number or a string
+                let provided = match provided {
+                    PostHogFlagFilterPropertyValue::Number(provided) => *provided,
+                    PostHogFlagFilterPropertyValue::String(provided) => {
+                        let Ok(provided) = provided.parse::<f64>() else {
+                            return Err(PostHogEvaluationError::Internal(
+                                "Can not parse the left side of the condition as a number",
+                            ));
+                        };
+                        provided
+                    }
+                    _ => {
+                        return Err(PostHogEvaluationError::Internal(
+                            "The left side of the condition is not a number or a string",
+                        ));
+                    }
+                };
+                match operator {
+                    "lt" => Ok(provided < requested),
+                    "gt" => Ok(provided > requested),
+                    _ => Err(PostHogEvaluationError::Internal("Unreachable code path")),
+                }
+            }
+            _ => Err(PostHogEvaluationError::Internal("Unsupported operator")),
+        }
+    }
+
+    /// Evaluate a percentage.
+    fn evaluate_percentage(&self, mapped_user_id: f64, percentage: i64) -> bool {
+        mapped_user_id <= percentage as f64 / 100.0
+    }
+
+    /// Evaluate a filter group for a feature flag. Returns `None` if the group is not matched or if there are errors
+    /// during the evaluation.
+    ///
+    /// Return values:
+    /// Ok(Some(variant)): matched and evaluated to this value
+    /// Ok(None): condition unmatched and not evaluated
+    fn evaluate_group(
+        &self,
+        group: &PostHogLocalEvaluationFlagFilterGroup,
+        hash_on_group_rollout_percentage: f64,
+        provided_properties: &HashMap<String, PostHogFlagFilterPropertyValue>,
+    ) -> Result<GroupEvaluationResult, PostHogEvaluationError> {
+        if let Some(ref properties) = group.properties {
+            for property in properties {
+                if let Some(value) = provided_properties.get(&property.key) {
+                    // The user provided the property value
+                    if !self.evaluate_condition(
+                        property.operator.as_ref(),
+                        value,
+                        &property.value,
+                    )? {
+                        return Ok(GroupEvaluationResult::Unmatched);
+                    }
+                } else {
+                    // We cannot evaluate, the property is not available
+                    return Err(PostHogEvaluationError::NotAvailable(
+                        "The required property in the condition is not available",
```

Comment:
nit: it's already fine (and much better than just `return None`), but adding some context to error messages would increase debugability (e.g. the name of the missing property).
It's applies to all other places as well